services/uploads.js

```diff
@@ -113,19 +113,29 @@ Uploads.createUploadCredentials = function() {
  */
 Uploads.createUpload = function(config) {
   v.assertShape({
-    mapId: v.required(v.string),
+    tileset: v.required(v.string),
     url: v.required(v.string),
-    tilesetName: v.string
+    name: v.string,
+    private: v.boolean
   })(config);
 
+  var body = {
+    tileset: config.tileset,
+    url: config.url
+  };
+
+  if (config.name) body.name = config.name;
+
+  if (config.private === false) {
+    body.private = false;
+  } else {
+    body.private = true;
```

Comment:
See https://github.com/mapbox/mapbox-sdk-js/blob/fc74a74f6c48d7373aec06548379f0c534f4d298/services/tilesets.js#L262-L277 for an example of a boolean option. You should just be able to pick out private without needing this.

Reply:
The `private` key defaults to `true`. The `pick` function won't default the boolean value to true. I could write my own `filter` function, but I think it would be more verbose instead of the logic that I have now. Unless I'm hardcoding names in the `filter` function, I would have to do some type of type checking to default `private` to `true`. Thoughts on this?